app/assets/javascripts/git_import.js

```diff
@@ -0,0 +1,48 @@
+/* global miqSparkleOn miqSparkleOff showErrorMessage clearMessages */
+
+var GitImport = {
+  retrieveDatastoreClickHandler: function() {
+    $('.git-retrieve-datastore').click(function(event) {
+      event.preventDefault();
+      miqSparkleOn();
+      clearMessages();
+
+      $.post('retrieve_git_datastore', $('#retrieve-git-datastore-form').serialize(), function(data) {
+        var parsedData = JSON.parse(data);
+        var messages = parsedData.message;
+        if (messages && messages.level === "error") {
+          showErrorMessage(messages.message);
+          miqSparkleOff();
+        } else {
+          GitImport.pollForGitTaskCompletion(parsedData);
+        }
+      });
+    });
+  },
+
+  pollForGitTaskCompletion: function(gitData) {
+    $.get('check_git_task', gitData, function(data) {
+      var parsedData = JSON.parse(data);
+      if (parsedData.state) {
+        setTimeout(GitImport.pollForGitTaskCompletion, 1500, gitData);
```

Comment:
@eclarizio - It might be nice to set a constant for this time out value. With that value's importance to this fix - the `1500` is hard to find buried in this function.